File: MiniDOMFinanziaria/Finanziaria/Prodotti/CategorieProdotto/CCategorieProdottoCursor.vb
Imports minidom.Databases
Imports minidom.Sistema

Partial Public Class Finanziaria

    ''' <summary>
    ''' Cursore sulla tabella delle Categorie Prodotto
    ''' </summary>
    ''' <remarks></remarks>
    <Serializable>
    Public Class CCategorieProdottoCursor
        Inherits DBObjectCursor(Of CCategoriaProdotto)

        Private m_Nome As New CCursorFieldObj(Of String)("Nome")
        Private m_NomeGruppo As New CCursorFieldObj(Of String)("NomeGruppo")
        Private m_Flags As New CCursorField(Of Integer)("Flags")
        Private m_DataInizio As New CCursorField(Of DateTime)("DataInizio")
        Private m_DataFine As New CCursorField(Of DateTime)("DataFine")

        Public Sub New()

        End Sub


        Public ReadOnly Property Nome As CCursorFieldObj(Of String)
            Get
                Return Me.m_Nome
            End Get
        End Property

        Public ReadOnly Property NomeGruppo As CCursorFieldObj(Of String)
            Get
                Return Me.m_NomeGruppo
            End Get
        End Property

        Public ReadOnly Property Flags As CCursorField(Of Integer)
            Get
                Return Me.m_Flags
            End Get
        End Property

        Public ReadOnly Property DataInizio As CCursorField(Of DateTime)
            Get
                Return Me.m_DataInizio
            End Get
        End Property

        Public ReadOnly Property DataFine As CCursorField(Of DateTime)
            Get
                Return Me.m_DataFine
            End Get
        End Property

        Public Overrides Function InstantiateNew(dbRis As DBReader) As Object
            Return New CCategoriaProdotto
        End Function

        Public Overrides Function GetTableName() As String
            Return "tbl_CQSPDCatProd"
        End Function

        Protected Overrides Function GetModule() As CModule
            Return Finanziaria.CategorieProdotto.Module
        End Function

        Protected Overrides Function GetConnection() As CDBConnection
            Return Finanziaria.Database
        End Function

    End Class


End Class

File: MiniDOMFinanziaria/Finanziaria/Prodotti/CategorieProdotto/CCategorieProdottoCursor.cs
using System;

namespace minidom
{
    public partial class Finanziaria
    {

        /// <summary>
    /// Cursore sulla tabella delle Categorie Prodotto
    /// </summary>
    /// <remarks></remarks>
        [Serializable]
        public class CCategorieProdottoCursor : Databases.DBObjectCursor<CCategoriaProdotto>
        {
            private DBCursorStringField m_Nome = new DBCursorStringField("Nome");
            private DBCursorStringField m_NomeGruppo = new DBCursorStringField("NomeGruppo");
            private DBCursorField<int> m_Flags = new DBCursorField<int>("Flags");
            private DBCursorField<DateTime> m_DataInizio = new DBCursorField<DateTime>("DataInizio");
            private DBCursorField<DateTime> m_DataFine = new DBCursorField<DateTime>("DataFine");

            public CCategorieProdottoCursor()
            {
            }

            public DBCursorStringField Nome
            {
                get
                {
                    return m_Nome;
                }
            }

            public DBCursorStringField NomeGruppo
            {
                get
                {
                    return m_NomeGruppo;
                }
            }

            public DBCursorField<int> Flags
            {
                get
                {
                    return m_Flags;
                }
            }

            public DBCursorField<DateTime> DataInizio
            {
                get
                {
                    return m_DataInizio;
                }
            }

            public DBCursorField<DateTime> DataFine
            {
                get
                {
                    return m_DataFine;
                }
            }

            public override object InstantiateNew(DBReader dbRis)
            {
                return new CCategoriaProdotto();
            }

            public override string GetTableName()
            {
                return "tbl_CQSPDCatProd";
            }

            protected override Sistema.CModule GetModule()
            {
                return CategorieProdotto.Module;
            }

            protected override Databases.CDBConnection GetConnection()
            {
                return Database;
            }
        }
    }
}